benchflow-experiment-manager/application/src/main/java/cloud/benchflow/experimentmanager/tasks/start/StartTask.java

```diff
@@ -77,8 +72,8 @@ public void run() {
 
       int nTrials = experiment.configuration().terminationCriteria().get().experiment().number();
 
-      // save experiment model in DB
-      experimentModelDAO.addExperiment(experimentID);
+      // save number of trials
+      experimentModelDAO.setNumTrials(experimentID, nTrials);
```

Comment:
This assumes you always know the number of trials beforehand, that might not be the case.